proxy_test.go

```diff
@@ -500,6 +542,42 @@ func TestReverseProxy_ServeHTTP1(t *testing.T) {
 				return makeCustomRequest(p, req)
 			},
 		},
+		{
+			cfg:           wildcardedCfg,
+			name:          "wildcarded Ok2",
+			expResponse:   "user: john-UK, password: john_pass\n" + okResponse,
+			expStatusCode: http.StatusOK,
+			f: func(p *reverseProxy) *http.Response {
+				req := httptest.NewRequest("POST", fakeServer.URL, nil)
+				req.Header.Set("X-ClickHouse-User", "john-UK")
+				req.Header.Set("X-ClickHouse-Key", "john_pass")
+				return makeCustomRequest(p, req)
+			},
+		},
+		{
+			cfg:           fullWildcardedCfg,
+			name:          "wildcarded Ok3",
+			expResponse:   "user: toto, password: titi\n" + okResponse,
+			expStatusCode: http.StatusOK,
+			f: func(p *reverseProxy) *http.Response {
+				req := httptest.NewRequest("POST", fakeServer.URL, nil)
+				req.Header.Set("X-ClickHouse-User", "toto")
+				req.Header.Set("X-ClickHouse-Key", "titi")
+				return makeCustomRequest(p, req)
+			},
+		},
+		{
+			cfg:           fullWildcardedCfg,
+			name:          "wildcarded Ko for default user",
+			expResponse:   "invalid username or password for user \"default\"",
+			expStatusCode: http.StatusUnauthorized,
+			f: func(p *reverseProxy) *http.Response {
+				req := httptest.NewRequest("POST", fakeServer.URL, nil)
+				req.Header.Set("X-ClickHouse-User", "default")
```

Comment:
Small nitpick: We should extract the header names to const to avoid typo's in the tests.